ngx-tools/src/jwt-token-managment/guards/defaultTokenRequired.ts

```diff
@@ -0,0 +1,36 @@
+import { Injectable } from '@angular/core';
+import { CanActivate, Params } from '@angular/router';
+import { select, Store } from '@ngrx/store';
+import { Observable } from 'rxjs';
+import { tap, filter, map, withLatestFrom } from 'rxjs/operators';
+import { getDefaultToken, getJwtTokenRoot } from '../selectors';
+import { FailedToActivateRoute } from '../actions';
+
+@Injectable()
+export class DefaultTokenRequired implements CanActivate {
+  currentLoadState = this.store.pipe(
+    select(getJwtTokenRoot()),
+    map((s) => (s && s.jwtTokens.initialTokenStatus) || 'uninitialized'),
+  );
+
+  currentToken = this.store.pipe(select(getDefaultToken));
+
+  constructor(
+    public store: Store<any>,
+  ) {}
+
+  canActivate(): Observable<boolean> {
+    return this.currentLoadState.pipe(
+      filter((s) => s !== 'uninitialized'),
+      withLatestFrom(this.currentToken),
+      map(([_, token ]) => token.length > 0),
+      tap((result) => {
+        if (!result) {
+          this.store.dispatch(new FailedToActivateRoute());
```

Comment:
are we expected to store the route somewhere so we can redirect users back to the original router after a successful login authentication?

Reply:
That is why I didn't implement the redirect myself, what you want to do with the failure due to a missing token is up to you. The README changes include an example, and I'll have a an example up for Engage shortly.

@ChrisKittredge Implemented this behavior in MyT.

Reply:
ah. ok.